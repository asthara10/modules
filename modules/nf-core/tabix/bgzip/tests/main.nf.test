nextflow_process {

    name "Test Process TABIX_BGZIP"
    script "../main.nf"
    config "./nextflow.config"
    process "TABIX_BGZIP"

    tag "modules"
    tag "modules_nfcore"
    tag "tabix"
    tag "tabix/bgzip"

    test("test_tabix_bgzip_compress") {


        when {
            process {
                """
                
                input[0] = [ [ id:'test' ], // meta map
              [ file(params.test_data['sarscov2']['illumina']['test_vcf'], checkIfExists: true) ]
            ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_tabix_bgzip_compress_gzi") {
        when {
            process {
                """
                        input[0] = [ [ id:'test' ], // meta map
              [ file(params.test_data['sarscov2']['illumina']['test_vcf'], checkIfExists: true) ]
            ]


                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }
    }
    
    test("test_tabix_bgzip_decompress") {
        when {
            process {
                """
                        input[0] = [ [ id:'test' ], // meta map
              [ file(params.test_data['homo_sapiens']['genome']['genome_bed_gz'], checkIfExists: true) ]
            ]


                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }
    }
    
}
