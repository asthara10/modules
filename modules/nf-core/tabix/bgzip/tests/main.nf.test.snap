{
    "versions": {
        "content": [
            [
                "versions.yml:md5,4e6c8368c84cdcf0a3769c7e382c9f8e"
            ]
        ],
        "timestamp": "2023-11-24T12:45:34.126026"
    }
}